### ISSUE TITLE
Autofix for rest/spread doesn't handle commas correctly

### ISSUE THREAD

**Issue by graingert** [mysticatea/eslint-plugin#12]
```js
  const eventPayload = {
    ...{
      category: 'Feedback-button',
      value,
      action: value > 0 ? 'upvote' : 'downvote',
    },
    ...analyticsData,
  };
```

Becomes

```js
  const eventPayload = {
    
      category: 'Feedback-button',
      value,
      action: value > 0 ? 'upvote' : 'downvote',
    ,
    ...analyticsData,
  };
```

which is a syntax error

**Comment by graingert** [comment]
@mysticatea :fire:

**Comment by graingert** [comment]
Also has a very bad interaction with: https://github.com/bryanrsmith/eslint-plugin-prefer-object-spread/pull/47 

where:
```js
return Object.assign({}, value, { cookie });
```
becomes
```js
return ({ ...({}), ...(value), ...({ cookie }) });
```
becomes
```js
return ({ , ...(value), ...({ cookie }) });
```

**Comment by mysticatea** [comment]
Thank you for this issue.

I confirmed it.

**Comment by graingert** [comment]
Neato

**Comment by mysticatea** [comment]
I released [v4.2.1](https://github.com/mysticatea/eslint-plugin/releases/tag/v4.2.1). Thank you!
